File: ptmgmt_api/app/models/attorney.rb
class Attorney < ApplicationRecord
    has_many :patients
    validates_presence_of :name, :address, :city, :state, :zip, :phone, :fax
    validates :name, :address, :phone, :fax, uniqueness: true
    validates :zip, length: { is: 5 }
    validates :state, length: { is: 2 }
end

File: ptmgmt_client/src/components/Attorney.js
import { useParams, useNavigate } from 'react-router-dom'
import { useSelector } from 'react-redux'
import { selectAttorneyById } from '../reducers/attorneyReducer'

function Attorney() {
  const id = parseInt(useParams().id)
  const attorney = useSelector(state => selectAttorneyById(state, id))
  const navigate = useNavigate()

  return (
    <div class="main">
      <div class="card">
        <h2>{attorney.name}</h2>
        <b>Address:</b><br/> {attorney.address}<br/>
        {attorney.city}, {attorney.state} {attorney.zip}<br/><br/>
        <b>Phone:</b><br/> {attorney.phone}<br/><br/>
        <b>Fax: </b><br/> {attorney.fax}<br/>

        <br/>
        <button onClick={() => navigate(-1)} >Back</button>
      </div>
    </div>
  )
};

export default Attorney